diff -Naur ./makefile.in ./makefile.in
--- ./makefile.in	2019-09-13 15:32:26.655742415 +0200
+++ ./makefile.in	2019-09-13 15:36:42.514572792 +0200
@@ -16,8 +16,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 
 CC = @CC@
-CFLAGS = @CFLAGS@
-CFLAGS_TRIANGLE = @CFLAGS_TRIANGLE@
+CFLAGS = @CFLAGS@ -shared -fPIC
+CFLAGS_TRIANGLE = @CFLAGS_TRIANGLE@ -shared -fPIC
 CFLAGS_VULNERABLE = @CFLAGS_VULNERABLE@
 LDFLAGS = @LDFLAGS@
 
@@ -72,7 +72,7 @@
 nnbathy: libnn.a nnbathy.c
 	$(CC) -o nnbathy nnbathy.c $(CFLAGS) -I. $(LDFLAGS) libnn.a $(MLIB)
 
-standalone: override LDFLAGS+=-static
+standalone: override LDFLAGS+=-shared
 standalone: $(PROGRAMS)
 	strip $(PROGRAMS)
 
@@ -83,7 +83,7 @@
 nncommon-vulnerable.o: override CFLAGS+=$(CFLAGS_VULNERABLE)
 
 triangle.o: triangle.c
-	$(CC) -c -DTRILIBRARY $(CFLAGS_TRIANGLE) -I. triangle.c
+	$(CC) -DTRILIBRARY $(CFLAGS_TRIANGLE) -I. triangle.c -o triangle.o
 
 tests: libnn.a $(TESTS)
 
