--- /tmp/portage/net-mail/kopanocore-8.6.81/work/kopanocore-8.6.81/Makefile.am	2018-10-12 12:15:14.000000000 +0200
+++ 1/Makefile.am	2019-10-23 11:07:32.427959085 +0200
@@ -836,7 +836,7 @@ SOAP_OUT = \
 ${SOAP_OUT}: provider/soap/KCmd.nsmap
 provider/soap/KCmd.nsmap: ${SOAPHEADER}
 	${MKDIR_P} provider/soap
-	soapcpp2 -jxw -Iinclude/ -d provider/soap/ $<
+	soapcpp2 -jxw -Iinclude/ -d provider/soap/ -Igsoap:/usr/share/gsoap/import/ $<
 
 BUILT_SOURCES += provider/soap/KCmd.nsmap ${SOAP_OUT}
 CLEANFILES += ${SOAP_OUT} provider/soap/*
--- /tmp/portage/net-mail/kopanocore-8.6.81/work/kopanocore-8.6.81/configure.ac	2019-09-05 13:01:39.000000000 +0200
+++ 1/configure.ac	2019-10-23 10:20:28.730286601 +0200
@@ -391,7 +391,7 @@ PKG_CHECK_MODULES([TCMALLOC], [libtcmall
 CPPFLAGS="$CPPFLAGS $TCMALLOC_CFLAGS"
 AC_CHECK_HEADERS([gperftools/malloc_extension.h google/malloc_extension.h])
 CPPFLAGS="$CPPFLAGS_system"
-PKG_CHECK_MODULES([VMIME], [vmime >= 0.9.2k1])
+PKG_CHECK_MODULES([VMIME], [vmime >= 0.9.2])
 PKG_CHECK_MODULES([jsoncpp], [jsoncpp >= 1.4.0], [], [
 	PKG_CHECK_MODULES([jsoncpp], [jsoncpp >= 0.8 jsoncpp < 1])
 ])
